analyses/0014-vitess/vitess-analysis.md

```diff
@@ -0,0 +1,1019 @@
+Documentation rubric.
+
+#### Information architecture
+
+**Conceptual content**: Is there high level conceptual (“About”) content?
+
+Yes, the site contains conceptual material explaining what Vitess is and how it
+works, in both the [Overview](https://vitess.io/docs/21.0/overview/) and
+[Concepts](https://vitess.io/docs/21.0/concepts/) sections.
+
+The introduction to the Overview could be more helpful to new users.
+
+**Feature complete**: Is the documentation feature complete? (i.e., each product
+feature is documented)?
+
+Yes, the documentation seems to cover all features of Vitesss (as far as I can
+tell). However, see the following regarding task instructions.
+
+**Task instructions**: Are there step-by-step instructions (tasks, tutorials)
+documented for features?
+
+Yes, there are task instructions covering all major classes of tasks required to
+use Vitess:
+
+- [Installation](https://vitess.io/docs/21.0/get-started/)
+- Setup and use
+- Administration
+
+Except for Installation, all task documentation is in the
+[User Guides](https://vitess.io/docs/21.0/user-guides/).
+
+Despite being labeled "Task-based guides," the task instructions in the User
+Guide are written from a feature-oriented perspective, which lessens the
+documentation's efficiency.
+
+**No features missing tasks**: Is the documentation free of any key features
+which are documented but missing task documentation?
+
+It looks like there are features that don't have adequate task descriptions.
+Features are represented, but often the task descriptions are not written as
+procedures. For examples, see
+[Information architecture](#information-architecture-1) in
+[Recommendations](#recommendations).
+
+**Happy path**: Is the “happy path”/most common use case documented?
+
+Vitess is a complex system with many configuration options. That said, I
+believe that setting up and running a production server in Kubernetes is the
+main use case, and is covered.
+
+**Isolated, atomic tasks**: Does task and tutorial content demonstrate atomicity
+and isolation of concerns? (Are tasks clearly named according to user goals?)
+
+No, tasks are not isolated. The User Guide has the following issues:
+
+- Tasks are intermingled with discussion of the features
+- Multiple tasks are presented per page
+- Task names are often not present in table of contents (TOC) headings
+- Where present, task names are ambiguous or misleading
+
+Titles of pages and sections are often feature-based, making it hard to find
+procedures to complete tasks.
+
+**Escalation path**: If the documentation does not suffice, is there a clear
+escalation path for users needing more help? (FAQ, Troubleshooting)
+
+No clear escalation path exists. There are Troubleshooting guides in the
+following sections:
+
+- Running in Production
+- Advanced > Distributed Atomic Transactions
+- Migration
+
+There is one special-purpose FAQ, for VReplication in the Reference.
```

Comment:
We do have this FAQ as well
https://vitess.io/docs/faq/

Reply:
Ah, OK. It's outside the versioned documentation so I missed it. There's a lot of good information here; I'll have a look and revise the analysis and recommendations!

Reply:
@deepthi , @frouioui : I've updated the analysis to account fro the FAQ. Apologies for not catching the sections outside the versioned documentation. Please review.